src/main/java/com/openshift/internal/restclient/model/Pod.java

```diff
@@ -61,10 +74,38 @@ public String getHost() {
 		}
 		return images;
 	}
+	
+	/**
+	 * The logic of the method is a copied from 'podStatus' function of
+	 * [app/scripts/filters/resources.js] of [openshift/origin-web-console]
+	 */
 
 	@Override
 	public String getStatus() {
-		return asString(POD_STATUS);
+		if (get(POD_DELETION_TIMESTAMP).isDefined()) {
```

Comment:
There is a class called ModelNodeBuilder where you could build and manipulate the values you desire and instantiate a Pod directly.  This would allow you to make changes without needing to craft permutations of the JSON string